pkg/otlp/internal/serializerexporter/exporter_test.go

```diff
@@ -0,0 +1,200 @@
+		setConfig      func(t *testing.T)
+		wantSketchTags tagset.CompositeTags
+		wantSerieTags  tagset.CompositeTags
+	}{
+		{
+			name: "no metrics",
+			genMetrics: func(t *testing.T) pmetric.Metrics {
+				return pmetric.NewMetrics()
+			},
+			wantSketchTags: tagset.NewCompositeTags([]string{}, nil),
+			wantSerieTags:  tagset.NewCompositeTags([]string{}, nil),
+		},
+		{
+			name: "no tags",
+			genMetrics: func(t *testing.T) pmetric.Metrics {
+				h := pmetric.NewHistogramDataPoint()
+				h.SetBucketCounts([]uint64{100})
+				h.SetCount(100)
+				h.SetSum(0)
+
+				n := pmetric.NewNumberDataPoint()
+				n.SetIntVal(777)
+				return newMetrics(histogramMetricName, h, numberMetricName, n)
+			},
+			wantSketchTags: tagset.NewCompositeTags([]string{}, nil),
+			wantSerieTags:  tagset.NewCompositeTags([]string{}, nil),
+		},
+		{
+			name: "metric tags and extra tags",
+			genMetrics: func(t *testing.T) pmetric.Metrics {
+				h := pmetric.NewHistogramDataPoint()
+				h.SetBucketCounts([]uint64{100})
+				h.SetCount(100)
+				h.SetSum(0)
+				hAttrs := h.Attributes()
+				hAttrs.Insert("histogram_1_id", pcommon.NewValueString("value1"))
+				hAttrs.Insert("histogram_2_id", pcommon.NewValueString("value2"))
+				hAttrs.Insert("histogram_3_id", pcommon.NewValueString("value3"))
+
+				n := pmetric.NewNumberDataPoint()
+				n.SetIntVal(777)
+				nAttrs := n.Attributes()
+				nAttrs.Insert("gauge_1_id", pcommon.NewValueString("value1"))
+				nAttrs.Insert("gauge_2_id", pcommon.NewValueString("value2"))
+				nAttrs.Insert("gauge_3_id", pcommon.NewValueString("value3"))
+				return newMetrics(histogramMetricName, h, numberMetricName, n)
+			},
+			setConfig: func(t *testing.T) {
+				config.Datadog.SetDefault("eks_fargate", true)
+				config.Datadog.SetDefault("tags", []string{"serverless_tag1:test1", "serverless_tag2:test2", "serverless_tag3:test3"})
+				t.Cleanup(func() {
+					config.Datadog.SetDefault("eks_fargate", false)
+					config.Datadog.SetDefault("tags", []string{})
+				})
+			},
+			wantSketchTags: tagset.NewCompositeTags(
+				[]string{
+					"serverless_tag1:test1", "serverless_tag2:test2", "serverless_tag3:test3",
+					"histogram_1_id:value1", "histogram_2_id:value2", "histogram_3_id:value3",
+				},
+				nil,
+			),
+			wantSerieTags: tagset.NewCompositeTags(
+				[]string{
+					"serverless_tag1:test1", "serverless_tag2:test2", "serverless_tag3:test3",
+					"gauge_1_id:value1", "gauge_2_id:value2", "gauge_3_id:value3",
+				},
+				nil,
+			),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.setConfig != nil {
+				tt.setConfig(t)
+			}
+			rec := &metricRecorder{}
+			exp, err := newExporter(
+				zap.NewNop(),
+				rec,
+				NewFactory(rec).CreateDefaultConfig().(*exporterConfig),
+			)
+			if err != nil {
+				t.Errorf("newExporter() returns unexpected error: %v", err)
+				return
+			}
+			if err := exp.ConsumeMetrics(context.Background(), tt.genMetrics(t)); err != nil {
```

Comment:
@mx-psi 
Sorry for delay. Added the unit test for this PR.

From my curiosity, I have a question about [StartIteration()](https://github.com/DataDog/datadog-agent/blob/8f8601d965874330cb50fd108f85f5eed657af66/pkg/metrics/iterable_series.go#L113-L127) that [(e *exporter) ConsumeMetrics()](https://github.com/DataDog/datadog-agent/blob/8f8601d965874330cb50fd108f85f5eed657af66/pkg/otlp/internal/serializerexporter/exporter.go#L150-L162) depends on.

**Why does not [source()](https://github.com/DataDog/datadog-agent/blob/8f8601d965874330cb50fd108f85f5eed657af66/pkg/metrics/iterable_series.go#L120) wait for [sink()](https://github.com/DataDog/datadog-agent/blob/8f8601d965874330cb50fd108f85f5eed657af66/pkg/metrics/iterable_series.go#L124)?**

This question came to my mind when adding this unit test because [(c *serializerConsumer) flush()](https://github.com/DataDog/datadog-agent/blob/8f8601d965874330cb50fd108f85f5eed657af66/pkg/otlp/internal/serializerexporter/consumer.go#L106) sometime invokes [s.SendIterableSeries(serieSource)](https://github.com/DataDog/datadog-agent/blob/8f8601d965874330cb50fd108f85f5eed657af66/pkg/otlp/internal/serializerexporter/consumer.go#L119) before [seriesSink.Append(serie)](https://github.com/DataDog/datadog-agent/blob/8f8601d965874330cb50fd108f85f5eed657af66/pkg/otlp/internal/serializerexporter/consumer.go#L116). It would  be difficult to add unit tests for [(e *exporter) ConsumeMetrics()](https://github.com/DataDog/datadog-agent/blob/8f8601d965874330cb50fd108f85f5eed657af66/pkg/otlp/internal/serializerexporter/exporter.go#L150-L162). 

Actually, this unit test sometime tests [L146](https://github.com/DataDog/datadog-agent/blob/8f8601d965874330cb50fd108f85f5eed657af66/pkg/otlp/internal/serializerexporter/exporter_test.go#L146), sometime not since the above reason. I don't think it is a good unit test, so do you have any ideas?

Reply:
If we can skip testing for serie tags, the results of the test can easily be constant.

Reply:
@ogaca-dd could you have a look at this? What's the recommended way of testing `IterableSeries`?

Reply:
@ogaca-dd Thank you! I changed code that `source()` wait for `sink()`. https://github.com/DataDog/datadog-agent/pull/11747/commits/6093df6949d05ad34a8b3080ea40b879caa161fe